compiler/test/stdlib/number.test.gr

```diff
@@ -630,10 +630,8 @@ assert Number.sign(-10000) == -1
 assert Number.sign(22222) == 1
 assert Number.sign(BI.toNumber(-1t)) == -1
 assert Number.sign(BI.toNumber(1t)) == 1
-// TODO(#693): Replace with Infinity literal when it exists
-assert 1 / Number.sign(0.0) == Float64.toNumber(Float64.infinity)
-// TODO(#693): Replace with -Infinity literal when it exists
-assert 1 / Number.sign(-0.0) == Number.neg(Float64.toNumber(Float64.infinity))
+assert 1 / Number.sign(0.0) == Infinity
+assert 1 / Number.sign(-0.0) == -Infinity
```

Comment:
There's a bunch more in this file that would be great to use the literal (obviously not the tests _for_ `Number.infinity`). It's essentially all of the `Number.infinity` and `Number.neg(Number.infinity)`.

Reply:
I was going to mention that. It probably also makes sense to have `Number.infinity` and `Number.nan` use the literals internally rather than `0.0 / 0.0` and `1.0 / 0.0`.

Reply:
It's almost as if these should have been done with the feature 😉  Anyway, I'm working on updating the file now.

Reply:
@ospencer this has been removed from this PR because I'm going to remove `infinity` and `NaN` constants in the number module.